hw11_telnet_client/main.go

```diff
@@ -1,6 +1,102 @@
 package main
 
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+)
+
+var timeout time.Duration
+
+func init() {
+	flag.DurationVar(&timeout, "timeout", 20, "timeout duration")
+}
+
 func main() {
-	// Place your code here,
-	// P.S. Do not rush to throw context down, think think if it is useful with blocking operation?
+	flag.Parse()
+
+	in := &bytes.Buffer{}
+	out := os.Stdout
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	ctx2, stop2 := signal.NotifyContext(ctx, os.Interrupt)
+	defer stop2()
+
+	ctx3, stop3 := signal.NotifyContext(ctx, syscall.SIGQUIT)
+	defer stop3()
+
```

Comment:
```suggestion
	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGTERM, syscall.SIGINT, os.Interrupt)
	defer stop()

```
Можно объединить несколько сигналов на закрытие в одном контексте вот таким образом